UEFI/amd64/hex0.hex0

```diff
@@ -0,0 +1,415 @@
+# SPDX-FileCopyrightText: 2022 Andrius Štikonas <andrius@stikonas.eu>
+# SPDX-FileCopyrightText: 2022 Jeremiah Orians <jeremiah@pdp10.guru>
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+# Usage: hex0 file.hex0 file
+# Does not validate the arguments or check for success
+
+# Calling convention:
+# First four arguments are passed via registers rcx, rdx, r8, r9 (if they fit in 64-bits)
+# but we need to leave stack space
+# rax, rcx, rdx, r8, r9, r10 and r11 are volatile and can be changed by called function
+
+# Registers:
+#   r12 in/fin: input file name, later reused for handle
+#   r13 out/fout: output file name, later reused for handle
+#   r14 system->boot->open_protocol from UEFI, later reused for rootdir
+#   r15 image_handle from UEFI
+
+# hex0 algorithm mostly follows stage0-posix version but uses rdi instead of rbp
+
+# DOS MZ header
+4D 5A                           # Signature
+78 00                           # Number of bytes in the last page.
+01 00                           # Number of whole/partial pages
+00 00                           # Number of entries in the relocation table.
+04 00                           # Header size
+00 00                           # Minimum allocation
+00 00                           # Maximum allocation
+00 00                           # Relocatable segment address for SS
+00 00                           # Initial value for SP
+00 00                           # Checksum (I don't think is looked at)
+00 00                           # Initial value for IP (Seems ignored)
+00 00                           # Relocatable segment address for CS (Seems ignored)
+40 00                           # The (absolute) offset to the relocation table.
+00 00                           # Value used for overlay management. If zero, this is the main executable
+00 00 00 00 00 00 00 00         # Reserved in PE
+00 00                           # OEM identifier
+00 00                           # OEM info
+00 00 00 00 00 00 00 00 00 00   # The required reserved 20 bytes of NULLS
+00 00 00 00 00 00 00 00 00 00
+40 00 00 00                     # Starting address of the PE header
+
+# [0x80]
```

Comment:
This comment is lying. There are 64 (0x40) bytes before it, not 0x80.

Did not check the subsequent offset comments.

Reply:
Thanks for fixing in c72070654660727b28334026cbb365dcd02e8f6b :)